### PR TITLE
[#5] Set up network layer to execute requests

### PR DESCRIPTION
**Pull request by Tuubz**
Closes https://github.com/nimblehq/ic-flutter-taher-toby/issues/5

## What happened 👀

Confirm that an API request can be executed successfully, by setting up a network layer with dependency injection.
 
## Insight 📝

I added `get_it` & `injectable` dependencies for dependency injection:
- https://pub.dev/packages/injectable → Used for annotations
- https://pub.dev/packages/get_it → Used for injecting

**Note**: `CredentialRepository` shouldn't be called directly in `main.dart`, we will refactor this later. 

## Proof Of Work 📹

I added `REST_API_ENDPOINT="https://jsonplaceholder.typicode.com"` to `.env.staging` for testing 💡 

https://user-images.githubusercontent.com/18277915/226829126-ba068e93-a6c0-405b-a120-374f06c5baf5.mov

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -51,6 +55,8 @@ class MyApp extends StatelessWidget {
   }
 }
 
+final credentialRepository = CredentialRepositoryImpl(getIt.get<ApiService>());
```
Comment (Tuubz) [inline lib/main.dart:58]:
```suggestion
final credentialRepository = getIt.get<CredentialRepository>();
```

Comment (Tuubz) [reply lib/main.dart:58]: I refactored this in [here](https://github.com/nimblehq/ic-flutter-taher-toby/pull/50/files#:~:text=final%20_credentialRepository%20%3D%20getIt.get%3CCredentialRepository%3E()%3B) 🙏

**Review by Shayokh144 — APPROVED** [review]

File: lib/di/module/network_module.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import 'package:flutter_survey/env.dart';
+import 'package:injectable/injectable.dart';
+
+@module
+abstract class NetworkModule {
+  @Singleton(as: ApiService)
+  ApiServiceImpl provideApiService(DioProvider dioProvider) {
+    return ApiServiceImpl(
+      dioProvider.getDio(),
+      baseUrl: Env.restApiEndpoint,
```
Comment (luongvo) [inline lib/di/module/network_module.dart:12]: Why don't we keep an [`.env.sample`](https://github.com/nimblehq/flutter-templates/blob/develop/template/.env.sample) file to clarify how many credentials we should define in the app? The external devs can easily clone the sample file and put valid credentials to run the project.

File: lib/di/provider/dio_provider.dart [hunk 3]
```diff
@@ -34,6 +38,7 @@ class DioProvider {
       ..options.connectTimeout = 3000
       ..options.receiveTimeout = 5000
       ..options.headers = {headerContentType: defaultContentType}
+      ..options.baseUrl = Env.restApiEndpoint
```
Comment (luongvo) [inline lib/di/provider/dio_provider.dart:41]: I don't think we need this config as we already provided it in `ApiService`.

**Review by luongvo — COMMENTED** [review]
The rest lgtm

**Comment by luongvo** [comment]
@Tuubz CI fails

**Pull request merged**